From cabfa8c395140f3b9f070d5a50c6fecbb890d4dd Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Mon, 19 Jan 2009 12:40:46 +0800
Subject: [PATCH 037/249] ENGR00105319 vpu: update code to make it compatible
 with bionic libc

Update code to make it compatible with bionic libc

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_io.c   | 21 ++++++++++++++++++---
 vpu/vpu_io.h   |  6 +-----
 vpu/vpu_lib.h  | 11 ++++++-----
 vpu/vpu_util.c |  6 ++----
 vpu/vpu_util.h |  3 ++-
 5 files changed, 29 insertions(+), 18 deletions(-)

diff --git a/vpu/vpu_io.c b/vpu/vpu_io.c
index 9400801..877271b 100644
--- a/vpu/vpu_io.c
+++ b/vpu/vpu_io.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  */
@@ -45,6 +45,9 @@ static unsigned long vpu_reg_base;
 
 unsigned int system_rev;
 semaphore_t *vpu_semap;
+vpu_mem_desc bit_work_addr;
+vpu_mem_desc pic_para_addr;
+vpu_mem_desc user_data_addr;
 
 int _IOGetPhyMem(int which, vpu_mem_desc *buff);
 
@@ -150,7 +153,7 @@ int isVpuInitialized(void)
 	return val != 0;
 }
 
-static int get_system_rev()
+static int get_system_rev(void)
 {
 	FILE *fp;
 	char buf[1024];
@@ -209,11 +212,23 @@ inline unsigned long *reg_map(unsigned long offset)
 int IOSystemInit(void *callback)
 {
 	int ret;
+	char *shm_path, shm_file[256];
 
-	vpu_semap = vpu_semaphore_open("/dev/shm/vpu.shm");
+	shm_path = getenv("VPU_SHM_PATH");
+
+	if (shm_path == NULL)
+		strcpy(shm_file, "/dev/shm");	/* default path */
+	else
+		strcpy(shm_file, shm_path);
+
+	strcat(shm_file, "/");
+	strcat(shm_file, "vpu.shm");
+
+	vpu_semap = vpu_semaphore_open(shm_file);
 
 	if (vpu_semap == NULL) {
 		err_msg("Error: Unable to open vpu shared memory file\n");
+		err_msg("Please export VPU_SHM_PATH env for putting vpu shared memory file\n");
 		return -1;
 	}
 
diff --git a/vpu/vpu_io.h b/vpu/vpu_io.h
index 23e03d6..d301750 100644
--- a/vpu/vpu_io.h
+++ b/vpu/vpu_io.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  */
@@ -63,10 +63,6 @@ typedef struct iram_t {
 
 typedef void (*vpu_callback) (int status);
 
-vpu_mem_desc bit_work_addr;
-vpu_mem_desc pic_para_addr;
-vpu_mem_desc user_data_addr;
-
 int IOSystemInit(void *callback);
 int IOSystemShutdown(void);
 int IOGetPhyMem(vpu_mem_desc * buff);
diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index b8ed070..cd1b58c 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  */
@@ -522,12 +522,13 @@ typedef struct vpu_versioninfo {
 
 /*
  * Revision History:
- * v4.3.2 [2008.10.28] support loopback in MX51
- * v4.2.2 [2008.09.03] support encoder in MX51
+ * v4.4.3 [2009.01.19] support chromaInterleave of encoder on MX51
+ * v4.3.2 [2008.10.28] support loopback on MX51
+ * v4.2.2 [2008.09.03] support encoder on MX51
  * v4.1.2 [2008.08.22] update MX37 VPU firmware to V1.0.5
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(4, 4, 2)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(4, 4, 3)
 
 extern unsigned int system_rev;
 
@@ -557,7 +558,7 @@ MXC_REV(cpu_is_mx37);
 MXC_REV(cpu_is_mx51);
 
 RetCode vpu_Init(void *);
-void vpu_UnInit();
+void vpu_UnInit(void);
 RetCode vpu_GetVersionInfo(vpu_versioninfo * verinfo);
 
 RetCode vpu_EncOpen(EncHandle *, EncOpenParam *);
diff --git a/vpu/vpu_util.c b/vpu/vpu_util.c
index c01188d..d86d2af 100644
--- a/vpu/vpu_util.c
+++ b/vpu/vpu_util.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  */
@@ -17,7 +17,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <assert.h>
-#include <pthread.h>
 #include <fcntl.h>
 #include <unistd.h>
 #include <sys/mman.h>
@@ -70,6 +69,7 @@ RetCode LoadBitCodeTable(Uint16 * pBitCode, int *size)
 	fp = fopen(fw_name, "rb");
 	if (fp == NULL) {
 		err_msg("Error in opening firmware binary file\n");
+		err_msg("Please put bin file to /usr/lib folder or export VPU_FW_PATH env\n");
 		return RETCODE_FAILURE;
 	}
 
@@ -765,8 +765,6 @@ semaphore_t *vpu_semaphore_open(char *semaphore_name)
 		ftruncate(fd, sizeof(semaphore_t));
 		pthread_mutexattr_init(&psharedm);
 		pthread_mutexattr_setpshared(&psharedm, PTHREAD_PROCESS_SHARED);
-		pthread_condattr_init(&psharedc);
-		pthread_condattr_setpshared(&psharedc, PTHREAD_PROCESS_SHARED);
 		semap = (semaphore_t *) mmap(NULL, sizeof(semaphore_t),
 					     PROT_READ | PROT_WRITE, MAP_SHARED,
 					     fd, 0);
diff --git a/vpu/vpu_util.h b/vpu/vpu_util.h
index 7ec3524..8fd0c33 100644
--- a/vpu/vpu_util.h
+++ b/vpu/vpu_util.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  */
@@ -17,6 +17,7 @@
 #define _VPU_UTIL_H_
 
 #include <sys/types.h>
+#include <pthread.h>
 
 #include "vpu_reg.h"
 #include "vpu_lib.h"
-- 
1.8.0

